eo2js/src/commands/transpile.js

```diff
@@ -72,6 +83,58 @@ const needsRetranspile = function(source, transpiled) {
   return !fs.existsSync(transpiled) || fs.statSync(source).mtime > fs.statSync(transpiled).mtime
 }
 
+/**
+ * Get the program name from XMIR (handles both old 'program' and new 'object' format).
+ * In old format: <program name="com.eo2js.simple">
+ * In new format (EO 0.59.0): Name is derived from package meta and main object name
+ * @param {any} xmir - Parsed XMIR
+ * @return {string} - The program name (e.g. "com.eo2js.simple")
+ */
+const getProgramName = function(xmir) {
+  const root = getRoot(xmir)
+  // Old format: name attribute on program element
```

Comment:
@konard we don't use comments inside method body: https://blog.pragmaticengineer.com/a-comment-is-an-invitation-for-refactoring/